File: app/src/main/java/com/junerver/cloudnote/db/NoteUtils.java
package com.junerver.cloudnote.db;

import com.junerver.cloudnote.CloudNoteApp;
import com.junerver.cloudnote.db.dao.NoteEntityDao;
import com.junerver.cloudnote.db.entity.NoteEntity;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by junerver on 2016/9/10.
 */
public class NoteUtils {

    /**
     * 列出全部数据
     * @return
     */
    public static List<NoteEntity> list(){
        List<NoteEntity> noteEntities;
        noteEntities= CloudNoteApp.getNoteEntityDao()
                .queryBuilder()
                .orderDesc(NoteEntityDao.Properties.Id)
                .list();
        return noteEntities;
    }

    /**
     * 列出未同步的所有数据
     * @return
     */
    public static List<NoteEntity> listNotSync(){
        //找出数据库中所有同步标记为否的
        List<NoteEntity> noteEntities;
        noteEntities=CloudNoteApp.getNoteEntityDao()
                .queryBuilder()
                .where(NoteEntityDao.Properties.IsSync.eq(false))
                .build()
                .list();
        return noteEntities;
    }
}

File: app/src/main/java/com/junerver/cloudnote/db/NoteUtils.kt
package com.junerver.cloudnote.db

import com.junerver.cloudnote.db.entity.Note
import com.junerver.cloudnote.db.entity.NoteEntity
import org.litepal.LitePal
import org.litepal.extension.find
import org.litepal.extension.findAll
import org.litepal.extension.findFirst

/**
 * Created by junerver on 2016/9/10.
 * update : 2021年10月14日14:23:52
 */
object NoteUtils {
    /**
     * 列出全部数据 排除本地删除
     * @return
     */
    fun listAll(): List<NoteEntity> {
        return LitePal.where("isLocalDel = false").order("updatedTime desc").find()
    }

    /**
     * 列出未同步的所有数据
     *
     * @return
     */
    fun listNotSync(): List<NoteEntity> {
        return LitePal.where("isSync = false")
            .find()

    }

    //根据id从数据库查找
    fun queryNoteById(id: String): NoteEntity? {
        return LitePal.where("objId = '$id'").findFirst()
    }
}